activesupport/lib/active_support/callbacks.rb

```diff
@@ -826,6 +754,30 @@ def get_callbacks(name)
       def set_callbacks(name, callbacks)
         send "_#{name}_callbacks=", callbacks
       end
+
+      def default_false_terminator
+        Proc.new do |target, result_lambda|
+          terminate = true
+          catch(:abort) do
+            result = result_lambda.call if result_lambda.is_a?(Proc)
+            if CallbackChain.halt_and_display_warning_on_return_false && result == false
```

Comment:
@kaspth That's the same concern I expressed [in my initial comment](https://github.com/rails/rails/pull/21218#issuecomment-139924781):

> should default_false_terminator live inside ActiveSupport? Did you put it there just because it's used both by ActiveRecord and ActiveModel or do you think it's more generic than that?

"Logically", it should live inside AR and AM… in other words there should be **two different settings**, like:

```ruby
ActiveRecord.halt_and_display_warning_on_return_false = ...
ActiveModel.halt_and_display_warning_on_return_false = ...
```

The benefit of having the setting in ActiveSupport is that you can have it just once.

Do you think that's a strong enough argument? Or you'd rather have it "logically" duplicated?